frontc/clexer.mll

```diff
@@ -65,7 +65,7 @@ let underline_error (buffer : string) (start : int) (stop : int) =
   (
     (if start' > 0 then (String.sub buffer 0 start') else "")
     ^ "\027[4m"
-    ^ (if (stop' - start') <> 0
+    ^ (if (stop' - start') > 0 && (stop' - start') < len
```

Comment:
It is still not enough, I would suggest the following, which properly checks that the subinterval is contained inside of a string. 

```suggestion
    ^ (if 0 <= start' && start' < stop' && stop <= len
```